rmw_connextdds_common/src/common/rmw_context.cpp

```diff
@@ -94,26 +292,39 @@ rmw_connextdds_initialize_participant_qos(
     return RMW_RET_ERROR;
   }
 
-  if (ctx->participant_qos_override_policy ==
-    rmw_context_impl_t::participant_qos_override_policy_t::All &&
-    DDS_StringSeq_get_length(&ctx->initial_peers) > 0)
-  {
-    if (!DDS_StringSeq_copy(&dp_qos.discovery.initial_peers, &ctx->initial_peers)) {
-      RMW_CONNEXT_LOG_ERROR_SET("failed to copy initial peers sequence")
-      return RMW_RET_ERROR;
-    }
+  switch (ctx->participant_qos_override_policy) {
+    case rmw_context_impl_t::participant_qos_override_policy_t::All:
+    case rmw_context_impl_t::participant_qos_override_policy_t::Basic:
+      if (nullptr != ctx->discovery_options) {
+        const auto rc = rmw_connextdds_initialize_discovery_options(ctx, dp_qos);
+        if (RMW_RET_OK != rc) {
+          RMW_CONNEXT_LOG_ERROR("failed to initialize discovery options")
+          return RMW_RET_ERROR;
+        }
+      }
+      if (DDS_StringSeq_get_length(&ctx->initial_peers) > 0 &&
+        !DDS_StringSeq_copy(&dp_qos.discovery.initial_peers, &ctx->initial_peers))
+      {
+        RMW_CONNEXT_LOG_ERROR_SET("failed to copy initial peers sequence")
+        return RMW_RET_ERROR;
+      }
+      break;
+    default:
+      break;
   }
 
   return RMW_RET_OK;
 }
 
 rmw_ret_t
 rmw_context_impl_t::initialize_node(
-  const rmw_discovery_options_t * const discovery_options)
+  const rmw_discovery_options_t * const discovery_options_in)
```

Comment:
Good call, thanks!

Reply:
Yeah, not strictly necessary, but I found it helpful to make sure we didn't accidentally pass one where it should have been the other.